models/kernels/Nonlinear_Reservoir.hpp

```diff
@@ -107,14 +107,25 @@ class Nonlinear_Reservoir
     }	
 
     //Function to update the response of storage in meters to an influx and timestep.
-    void response_storage_meters(double in_flux_meters_per_second, int delta_time_seconds, double &excess_water_meters)
+    double response_storage_meters(double in_flux_meters_per_second, int delta_time_seconds, double &excess_water_meters)
     {	
+        double outlet_velocity_meters_per_second = 0;
+        double sum_of_outlet_velocities_meters_per_second = 0;
+
+        //Update current storage from influx multiplied by delta time.
         state.current_storage_height_meters += in_flux_meters_per_second * delta_time_seconds;
 	
+        //Loop through reservoir outlets.
         for (auto& outlet : this->outlets) //pointer to outlets
         {
+            //Calculate outlet velocity.
+            outlet_velocity_meters_per_second = outlet.velocity_meters_per_second(parameters, state);
+
             //Update storage from outlet velocity multiplied by delta time.
-            state.current_storage_height_meters -= outlet.velocity_meters_per_second(parameters, state) * delta_time_seconds;
+            state.current_storage_height_meters -= outlet_velocity_meters_per_second * delta_time_seconds;
```

Comment:
Really like the units in these variables, makes dimensional analysis pretty quick!